custom_components/adjustable_bed/coordinator.py

```
@@ -369,40 +399,41 @@ async def _select_adapter(self) -> AdapterSelectionResult:
-                        # Handle None RSSI by using a low default value
-                        rssi_value = svc_rssi if svc_rssi is not None else -999
-                        svc_source = getattr(svc_info, 'source', 'unknown')
-                        _LOGGER.debug(
-                            "Auto-select candidate: source=%s, rssi=%s",
-                            svc_source, rssi_value
-                        )
-                        if rssi_value > best_rssi:
-                            best_rssi = rssi_value
-                            best_source = svc_source
+                discovered_services = list(bluetooth.async_discovered_service_info(self.hass, connectable=True))
             except Exception as err:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)